common/lib/dependabot/file_fetchers/base.rb

```diff
@@ -684,6 +693,17 @@ def decode_binary_string(str)
         bom = (+"\xEF\xBB\xBF").force_encoding(Encoding::BINARY)
         Base64.decode64(str).delete_prefix(bom).force_encoding("UTF-8").encode
       end
+
+      def find_submodules(path)
+        SharedHelpers.run_shell_command(
+          <<~CMD
+            git -C #{path} ls-files --stage
+          CMD
+        ).split("\n").filter_map do |line|
+          info = line.split
+          next info.last if info.first == "160000"
```

Comment:
what is 160000? Is it a special number?

Reply:
Yeah, it's the permissions git assigns to a submodule. I'll add comments for this.

Reply:
I ended up adding https://github.com/dependabot/dependabot-core/pull/6718/commits/e2f040f7df958eff20588d5fb460a1a03fbf32a8, so no more magic number!